pkg/actuators/machine/actuator.go

```diff
@@ -362,9 +353,19 @@ func (a *Actuator) Update(context context.Context, cluster *clusterv1.Cluster, m
 	// Parent controller should prevent this from ever happening by calling Exists and then Create,
 	// but instance could be deleted between the two calls.
 	if len(instances) == 0 {
-		glog.Warningf("attempted to update machine but no instances found")
+		glog.Warningf("attempted to update machine but no running instances found")
+
+		// check for all non-terminated instances, not just running
+		existingInstances, err := a.getExistingMachineInstances(cluster, machine)
```

Comment:
I don't think it makes sense to update stopped instance.

Reply:
Hmm, this is confusing. If you know there are no running instance for `machine`, you could just call `getStoppedInstances` directly.